[docs] Classic/Standalone apps: Update the documentation (#18146)
expo/expo: docs/pages/classic/building-standalone-apps.md

      "package": "com.yourcompany.yourappname",
      "versionCode": 1
    }
  }
}
```

Here are the required fields that you have to pay attention to when configuring your **app.json** or **app.config.js** file:

- Add fields such as `name`, `icon`, and `version`.
- The iOS `bundleIdentifier` and Android `package` fields use reverse DNS notation but do not have to be related to a domain. Replace `"com.yourcompany.yourappname"` with whatever makes sense for your app.
- The `slug` uses a URL as its value. It is the combination of username and the actual slug that your app's JavaScript is published to. For example: `expo.dev/@community/native-component-list`, where `community` is the username and `native-component-list` is the slug.
- The `ios.buildNumber` and `android.versionCode` distinguish different binaries of your app. Make sure to increment these for each build you upload to the App Store or Google Play Store.

> There are other options you can add to **app.json**. For example, some developers like to configure their own build number, linking scheme, etc. We recommend you to go through [Configuration with app.json/app.config.js](/workflow/configuration) for the full specification. At this step, also check [Deploying to App Stores](/distribution/app-stores) documentation. It provides details on what type of metadata is required for App stores.

Author: Duly noted. I will add this to the style guide.

To build an APK, run the command:

<Terminal cmd={['$ expo build:android -t apk']} />

It is recommended to build an Android App Bundle (aab) for submitting the app to the Google Play Store. To build an aab, run the command:

<Terminal cmd={['$ expo build:android -t app-bundle']} />

When building abb, you have to make sure the [Google Play App Signing](/app-signing/app-credentials) is enabled for your project. You can read more about it in the [Android documentation](https://developer.android.com/guide/app-bundle).

#### Generating keystore

When building your app for the first time, you have to generate a [keystore](https://developer.android.com/training/articles/keystore). You can upload your own if you are familiar with process of generating keystore, or you can let Expo handle the process.

For Expo to handle the process, choose the first option in the command-line when prompted for keystore:

```sh
[exp] No currently active or previous builds for this project.

Would you like to upload a keystore or have us generate one for you?
If you don't know what this means, let us handle it! :)

1) Let Expo handle the process!
2) I want to upload my own keystore!
```

> If you choose the first option, it is **strongly recommend** that you run the following command after the build process is complete and backup the keystore to a safe location:

Reviewer: ```suggestion
If you choose the first option, it is **strongly recommended** that you run the following command after the build process is complete. After running it, backup the keystore in a safe location:
```

<Terminal cmd={['$ expo fetch:android:keystore']} />

Once you submit the app to the Google Play Store, all future updates to that app **must** be signed with the same keystore that the Play Store initially accepted. If you do not have a local copy of your keystore, you will be unable to publish new versions of your app to the Play Store.

> For any reason you delete your project or clear your credentials in the future, you will not be able to submit any updates to your app without providing the same keystore. Your only option left is to generate a new keystore and re-upload your application as a new application.

Reviewer: ```suggestion
If for any reason you delete your project or clear your credentials in the future, you will not be able to submit any updates to your app without providing the same keystore. If you're unable to retrieve the keystone, your only option left is to generate a new keystore and re-upload your application as a new application.
```

#### Uploading your keystore after publishing the app

If you choose the first option when generating the keystore and later decide to upload your own keystore, you can use the option to clear the current Android keystore from Expo's build servers by running the command:

<Terminal cmd={['$ expo build:android --clear-credentials']} />

**This is irreversible, so only run this command if you have generated your own keystore.** You can learn more about how code signing and keystores work in the [Android documentation](https://developer.android.com/studio/publish/app-signing.html).

### Building for iOS

There are two types of build you can generate when building a standalone app for iOS: archive and simulator.

> Archive build is used for publishing your app to the Apple App Store or distributing it with services like TestFlight. To build an archive, run the command:

Reviewer: ```suggestion
Archive builds are used for submitting your app to the Apple App Store and distributing it with features like TestFlight. To build an iOS app archive, run:
```

<Terminal cmd={['$ expo build:ios -t archive']} />

> Simulator build is used to test the standalone app on an iOS simulator. To build it, run the command:

Reviewer: ```suggestion
Simulator builds are used to test a standalone app in an iOS simulator. To build one, run:
```

<Terminal cmd={['$ expo build:ios -t simulator']} />

> Building for iOS, you can choose the option to let Expo create necessary credentials. You still have the choice to provide your own credentials. In this process, your Apple ID and password are used locally and are never saved on Expo's servers.

Reviewer: ```suggestion
When building for iOS, you can choose to let Expo create any necessary credentials that your app will need. However, you'll still have the choice to provide your own credentials if you'd like. During this process, Expo CLI will use your Apple ID and password. Your authentication information is only used locally on your computer and is never sent to Expo's servers.
```

  I will provide all the credentials and files needed, Expo does limited validation
```

> If you are unfamiliar with iOS credentials, it is recommended to let Expo handle the creation & management of credentials. To know more about them, refer to the [App Credentials documentation](/app-signing/app-credentials).

Reviewer: ```suggestion
If you are unfamiliar with iOS credentials, it is recommended to let Expo handle the creation & management of credentials. To learn more about them, refer to the [App Credentials documentation](/app-signing/app-credentials).
```

If you are familiar with iOS credentials and plan to provide your own, use the [Apple Developer Portal](https://developer.apple.com/account/resources/certificates/list) to create them.

#### Enterprise distribution

To build and distribute your app using Apple's Enterprise distribution, it is required that you are a member of the [Apple Developer Enterprise Program](https://developer.apple.com/programs/enterprise/).

> Expo Build Service supports Enterprise distribution. When you call `expo build:ios`, you just need to choose the correct team that is labeled with `(In-House)`.

Reviewer: ```suggestion
Expo's build service supports Enterprise distribution. When you run `expo build:ios`, you'll need to choose the correct team that is labeled with `(In-House)`.
```

#### Adhoc distribution

> > This distribution requires you to override and use your own credentials credential overrides. Use this distribution if you know how to generate iOS credentials.

Reviewer: ```suggestion
> This distribution method requires you to override your own credentials. Use this distribution if you know how to generate iOS credentials.
```

> One of the easiest ways to test your standalone app is using a [simulator build](#building-for-ios) and then using TestFlight for your physical device for testing. However, if you want to build and install an IPA directly on a physical device through Xcode, follow the steps below to generate one using Expo CLI:

Reviewer: ```suggestion
One of the easiest ways to test your standalone app is using a [simulator build](#building-for-ios) and then using TestFlight for your physical device for testing. However, if you want to build and install an app directly on a physical device through Xcode, follow the steps below to generate one using Expo CLI:
```
Another styleguide point. We should say "app" when referring to a .ipa .apk or .aab file. We should use ".ipa" when we want to specifically talk about the extension.

> - Generate preliminary certificate such as app identifier and distribution certificate by running the command:

Reviewer: ```suggestion
- Generate preliminary certificates such as app identifier and distribution certificate by running the command:
```

<Terminal cmd={['$ expo build:ios --provisioning-profile-path ~/Path/To/provisioningProfileYouCreated.mobileprovision']} />

- Make sure the other certificates you use match the ones you selected when creating your adhoc provisioning profile in the previous step.

After Expo finishes your build, you can install it onto your device via Xcode.

- Open the `Devices & Simulators` window in Xcode
- Select your connected device
- Under `Installed Apps`, click the plus (`+`) and then select the IPA generated by Expo

> **Note:** We enable bitcode for iOS, so the **.ipa** files for iOS are much larger than the eventual App Store download available to your users. For more information, see [App Thinning](https://help.apple.com/xcode/mac/current/#/devbbdc5ce4f).

### Switch to Push Notification Key on iOS

If you use a Push Notifications Certificate and want to switch to a [Push Notifications Key](app-signing/app-credentials/#push-notification-keys), start the build with `--clear-push-cert` option. Expo will remove legacy certificates from its servers and generate a Push Notifications Key for you.

## 4. Wait for it to finish building

> Expo Build Service uses building machines to build your app. The build process starts as soon as one of the machines is free and ready to be used. To check how long the waiting time for your build is, visit [Turtle status](https://expo.dev/turtle-status) site.

Reviewer: ```suggestion
Expo's build service uses cloud services to build your app. The build process starts as soon as one of the machines is free and ready to be used. To check how long the waiting time for your build is, visit [Turtle status](https://expo.dev/turtle-status) site.
```

> Expo CLI prints a URL with a unique ID, such as `expo.dev/builds/some-unique-id`, for you to visit and check your build progress. You can also access build logs on this URL.

Reviewer: ```suggestion
Expo CLI prints a URL that you can visit and check your build for its progress and logs.
```
